.gitmodules

```diff
@@ -58,3 +58,6 @@
 	path = cmake/configs/uavcan_board_ident
 	url = https://github.com/PX4/uavcan_board_ident.git
 	branch = master
+[submodule "src/platforms/posix/lockstep_scheduler"]
+	path = src/platforms/posix/lockstep_scheduler
+	url = git@github.com:julianoes/lockstep-scheduler.git
```

Comment:
Why is this worth keeping as a separate project?

Reply:
Happy to move that into PX4. It's just I needed to develop this with unit testing, and I have not figured out how to do that inside PX4 without the overhead of compiling everything SITL and then running it all.

Reply:
Copied in now.

Reply:
We need a simple way for unit testing. I end up hacking the whole SITL environment just to replace basic "unit tests" and then deleting everything before pushing such that the horrible hacks are out of the way again...

Reply:
@MaEtUgR definitely. I've been experimenting with catch2 on the side. 

The bigger problem we'll need to tackle is structuring modules and drivers so that they're unit testable to begin with. Either with dependency injection or in some cases breaking off the core functionality into a standalone library separated from the PX4 boilerplate.